packages/clerk-js/webpack.config.js

```diff
@@ -0,0 +1,187 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const webpack = require('webpack');
+const packageJSON = require('./package.json');
+const path = require('path');
+const { merge } = require('webpack-merge');
+const ReactRefreshWebpackPlugin = require('@pmmmwh/react-refresh-webpack-plugin');
+const ReactRefreshTypeScript = require('react-refresh-typescript');
+const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
+
+const isProduction = mode => mode === 'production';
+const isDevelopment = mode => !isProduction(mode);
+
+/** @type { () => import('webpack').Configuration } */
+const common = ({ mode }) => {
+  return {
+    mode,
+    resolve: {
+      extensions: ['.ts', '.tsx', '.js', '.jsx'],
+    },
+    plugins: [
+      new webpack.DefinePlugin({
+        __DEV__: isDevelopment(mode),
+        __PKG_VERSION__: JSON.stringify(packageJSON.version),
+        __PKG_NAME__: JSON.stringify(packageJSON.name),
+      }),
+      new webpack.EnvironmentPlugin({
+        CLERK_ENV: mode,
+        NODE_ENV: mode,
+      }),
+    ],
+  };
+};
+
+const svgLoader = () => {
+  return {
+    test: /\.svg$/,
+    use: ['@svgr/webpack'],
+  };
+};
+
+const typescriptLoaderProd = () => {
+  return {
+    test: /\.(ts|js)x?$/,
+    exclude: /node_modules/,
+    use: [
+      {
+        loader: 'ts-loader',
+        options: { transpileOnly: true },
+      },
+    ],
+  };
+};
+
+const typescriptLoaderDev = () => {
+  return {
+    test: /\.(ts|js)x?$/,
+    exclude: /node_modules/,
+    use: [
+      {
+        loader: 'ts-loader',
+        options: {
+          configFile: 'tsconfig.dev.json',
+          transpileOnly: true,
+          getCustomTransformers: () => ({
+            before: [ReactRefreshTypeScript()],
+          }),
+        },
+      },
+    ],
+  };
+};
+
+/** @type { () => (import('webpack').Configuration) } */
+const commonForProd = () => {
+  return {
+    devtool: undefined,
+    module: {
+      rules: [svgLoader(), typescriptLoaderProd()],
+    },
+    output: {
+      path: path.resolve(__dirname, 'dist'),
+      filename: '[name].js',
+      libraryTarget: 'umd',
+      globalObject: 'globalThis',
+    },
+  };
+};
+/** @type { () => (import('webpack').Configuration) } */
+const externalsForProd = () => {
```

Comment:
🙃 `externalsForProdHeadless` or `externalsForHeadless` ?

Reply:
I like this more, changed :)